services/ux-backend/handlers/devicefinder/handler.go

```diff
@@ -0,0 +1,174 @@
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		klog.Errorf("failed to decode request body: %v", err)
+		http.Error(w, "Invalid request body", http.StatusBadRequest)
+		return
+	}
+
+	// Use namespace from request or default
+	targetNamespace := req.Namespace
+	if targetNamespace == "" {
+		targetNamespace = defaultNamespace
+	}
+
+	// Create device finder daemonset
+	err := devicefinder.CreateDeviceFinderDaemonSet(r.Context(), cl, targetNamespace, req.NodeSelector, req.Tolerations)
+	if err != nil {
+		klog.Errorf("failed to create device finder daemonset: %v", err)
+		response := DiscoveryResponse{
+			Message: fmt.Sprintf("Failed to create device finder daemonset: %v", err),
+			Success: false,
+		}
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusInternalServerError)
+		json.NewEncoder(w).Encode(response)
+		return
+	}
+
+	// Return success response
+	response := DiscoveryResponse{
+		Message: "Device finder daemonset created successfully",
+		Success: true,
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(response)
+}
+
+func handleGet(w http.ResponseWriter, r *http.Request, cl client.Client, defaultNamespace string) {
+	// Parse request body
+	var req DiscoveryRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		klog.Errorf("failed to decode request body: %v", err)
+		http.Error(w, "Invalid request body", http.StatusBadRequest)
+		return
+	}
+
+	// Use namespace from request or default
+	targetNamespace := req.Namespace
+	if targetNamespace == "" {
+		targetNamespace = defaultNamespace
+	}
+
+	// List all configmaps with devicefinder-result- prefix
+	configMapList := &corev1.ConfigMapList{}
+	err := cl.List(r.Context(), configMapList, client.InNamespace(targetNamespace), client.MatchingLabels{"app": "devicefinder"})
+	if err != nil {
+		klog.Errorf("failed to list device configmaps: %v", err)
+		response := DiscoveryResponse{
+			Message: fmt.Sprintf("Failed to list device configmaps: %v", err),
+			Success: false,
+		}
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusInternalServerError)
+		json.NewEncoder(w).Encode(response)
+		return
+	}
+
+	// Parse configmaps and extract device information
+	devices := make(map[string]DiscoveryResult)
+	for _, configMap := range configMapList.Items {
+		// Extract node name from configmap name (format: devicefinder-result-<node-name>)
+		if len(configMap.Name) <= len("devicefinder-result-") {
+			continue
+		}
+		nodeName := configMap.Name[len("devicefinder-result-"):]
+
+		// Parse discovered devices from configmap data
+		var discoveredDevices []DiscoveredDevice
+		if devicesJSON, exists := configMap.Data["discovered-devices"]; exists {
+			if err := json.Unmarshal([]byte(devicesJSON), &discoveredDevices); err != nil {
+				klog.Errorf("failed to unmarshal discovered devices for node %s: %v", nodeName, err)
+				continue
```

Comment:
return internal server error

Reply:
Yes this should be fine, because if there a corruption of configmap we should not send half details to the user.